### PR TITLE
Subgraph v5 experimental

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: src/mappings/Controller/schema.graphql [hunk 1]
```diff
@@ -87,6 +89,13 @@ type ContributionRewardExtParam @entity {
 	rewarder: Bytes!
 }
 
+type SchemeFactoryParam @entity {
+	id: ID!
+	votingMachine: Bytes!
+	voteParams: GenesisProtocolParam!
+	daoFactory: Bytes!
```
Comment (orenyodfat) [inline src/mappings/Controller/schema.graphql]: why daoFactory is needed here ?

File: src/mappings/SchemeFactory/mapping.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+// Import entity types generated from the GraphQL schema
+import {
+   SchemeFactoryNewSchemeProposal,
+   SchemeFactoryProposal,
+   SchemeFactoryProposalExecuted,
+} from '../../types/schema';
+import { equalsBytes, equalStrings, eventId } from '../../utils';
+
+export function handleNewSchemeProposal(event: NewSchemeProposal): void {
+  let ent = SchemeFactoryNewSchemeProposal.load(eventId(event));
```
Comment (orenyodfat) [inline src/mappings/SchemeFactory/mapping.ts]: how about the id will be the proposalId?

File: src/mappings/Controller/schema.graphql (hunk 1 above)
Comment (ben-kaufman) [reply src/mappings/Controller/schema.graphql]: It's a parameter of the SchemeFactory. It's not necessary I guess but I can't l can't know what will be needed by the UI team and we usually index all parameters like that.

File: src/mappings/SchemeFactory/mapping.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                          ent.schemeName,
+                          ent.schemeData,
+                          ent.packageVersion as BigInt[],
+                          ent.permission,
+                          ent.schemeToReplace as Address);
+}
+
+export function handleProposalExecuted(event: ProposalExecuted): void {
+  let ent = SchemeFactoryProposalExecuted.load(eventId(event));
+  if (ent == null) {
```
Comment (orenyodfat) [inline src/mappings/SchemeFactory/mapping.ts]: is it not always null ?

File: src/mappings/SchemeFactory/mapping.ts (hunk 2 above)
Comment (ben-kaufman) [reply src/mappings/SchemeFactory/mapping.ts]: That's how we did it in the SchemeRegistrar and other schemes. We can change it I guess but didn't want to make it" stand out".

File: src/mappings/SchemeFactory/mapping.ts (hunk 3 above)
Comment (ben-kaufman) [reply src/mappings/SchemeFactory/mapping.ts]: I'm pretty sure you have to have these checks even when they seem futile, if I remember correctly that's required due to some Graph Node special requirements.

File: src/mappings/SchemeFactory/mapping.ts (hunk 2 above)
Comment (orenyodfat) [reply src/mappings/SchemeFactory/mapping.ts]: well:
1. event is unique and for that this line is redundant if we use event
2. I feel that proposalId is better

Comment (ben-kaufman) [reply src/mappings/SchemeFactory/mapping.ts]: ok...

File: src/mappings/SchemeFactory/mapping.ts (hunk 3 above)
Comment (orenyodfat) [reply src/mappings/SchemeFactory/mapping.ts]: if that is the case so it is an issue with the graph , which give same event id for multiple events.
using porposalId is better..

**Pull request merged**